test/grouping.jl

```diff
@@ -2691,4 +2691,122 @@ end
     @test map(nrow, gdf) == [1, 1, 1]
 end
 
+@testset "isagg fix" begin
+    for fun in (sum, prod, mean, var, std, sum∘skipmissing, prod∘skipmissing,
+                mean∘skipmissing, var∘skipmissing, std∘skipmissing),
+        col in ([1, 2, 3], [big(1.5), big(2.5), big(3.5)], [1 + 0.5im, 2 + 0.5im, 3 + 0.5im],
+                [true, false, true], [pi, pi, pi], [1//2, 1//3, 1//4],
+                Real[1, 1.5, 1//2], Number[1, 1.5, 1//2], Any[1, 1.5, 1//2],
+                [1, 2, missing], [big(1.5), big(2.5), missing], [1 + 0.5im, 2 + 0.5im, missing],
+                [true, false, missing], [pi, pi, missing], [1//2, 1//3, missing],
+                Union{Missing,Real}[1, 1.5, missing],
+                Union{Missing,Number}[1, 1.5, missing], Any[1, 1.5, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (maximum, minimum, maximum∘skipmissing, minimum∘skipmissing),
+        col in ([1, 2, 3], [big(1.5), big(2.5), big(3.5)],
+                [true, false, true], [pi, pi, pi], [1//2, 1//3, 1//4],
+                Real[1, 1.5, 1//2], Number[1, 1.5, 1//2], Any[1, 1.5, 1//2],
+                [1, 2, missing], [big(1.5), big(2.5), missing],
+                [true, false, missing], [pi, pi, missing], [1//2, 1//3, missing],
+                Union{Missing,Real}[1, 1.5, missing],
+                Union{Missing,Number}[1, 1.5, missing], Any[1, 1.5, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (first, last, length, first∘skipmissing, last∘skipmissing),
+        col in ([1, 2, 3], [big(1.5), big(2.5), big(3.5)], [1 + 0.5im, 2 + 0.5im, 3 + 0.5im],
+                [true, false, true], [pi, pi, pi], [1//2, 1//3, 1//4],
+                Real[1, 1.5, 1//2], Number[1, 1.5, 1//2], Any[1, 1.5, 1//2],
+                [1, 2, missing], [big(1.5), big(2.5), missing], [1 + 0.5im, 2 + 0.5im, missing],
+                [true, false, missing], [pi, pi, missing], [1//2, 1//3, missing],
+                Union{Missing,Real}[1, 1.5, missing],
+                Union{Missing,Number}[1, 1.5, missing], Any[1, 1.5, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        if fun isa typeof(last∘skipmissing)
+            # this is another hard corner case
```

Comment:
This is a special case, because it works in fast path, but fails in slow path. I have changed the comment

Reply:
Hmm, it works only with `GroupedDataFrame` input, not with a `DataFrame` input. I don't get why.

Also, I thought the goal of this PR was to make the slow and fast path behave the same, so shouldn't this always throw an error whatever the eltype?

Reply:
I did not see the benefit of `last∘skipmissing` throwing an error in cases when we can efficiently produce a correct result. The fact that `last∘skipmissing` errors in Base is only due to `O(1)` restriction in `last` docstring, e.g. `first∘skipmissing` works in base because its docstring does not require `O(1)`. I think we should change in Base that `last∘skipmissing` works.

> it works only with `GroupedDataFrame` input, not with a `DataFrame` input. I don't get why.

It should work in `combine` both for `GroupedDataFrame` and `DataFrame` and with `select` on `GroupedDataFrame` (and pseudo-broadcasting is applied in this case). It is expected to fail for `select` on `DataFrame`. Do you observe a different behaviour?

Reply:
Yeah, `last(skipmissing(x))` and `lastindex(skipmissing(x))` should probably be defined when `x` is an `AbstractArray`, since they don't require going over the whole collection.

> It should work in `combine` both for `GroupedDataFrame` and `DataFrame` and with `select` on `GroupedDataFrame` (and pseudo-broadcasting is applied in this case). It is expected to fail for `select` on `DataFrame`. Do you observe a different behaviour?

I get this
```julia
julia> df = DataFrame(x=[1])
1×1 DataFrame
│ Row │ x     │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │

julia> combine(df, :x => last ∘ skipmissing)
ERROR: MethodError: no method matching lastindex(::Base.SkipMissing{Array{Int64,1}})
```

Reply:
Yeah - I have forgotten of this path. So you have:
```
julia> df = DataFrame(x=[1])
1×1 DataFrame
│ Row │ x     │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │

julia> combine(df, :x => last ∘ skipmissing)
ERROR: MethodError: no method matching lastindex(::Base.SkipMissing{Array{Int64,1}})

julia> combine(:x => last ∘ skipmissing, df)
1×1 DataFrame
│ Row │ x_last_skipmissing │
│     │ Int64              │
├─────┼────────────────────┤
│ 1   │ 1                  │
```
which is unfortunate.

I would not touch it though now as it is not easy to fix.

In the long term we should rewrite the whole engine for `select`/`transform`/`combine` to be unified. Now - unfortunately - I have designed it in stages, and when initially implementing `select` and `transform` we have not envisioned that we will have a full ecosystem that we have now, so we essentially have two processing engines - one for `combine` in `GroupedDataFrame` and the other for `select` in `DataFrame`.

This should be implemented in a way that `select`/`transform`/`combine` on `DataFrame` are always essentially calls to `GroupedDataFrame` on a group formed by no columns (single group). Additional things to remember when implementing it:
* add better support for 0 groups
* add possibility for a group to have zero rows
* add requested extensions to the syntax of `source_columns => function => target_col_name`

Reply:
OK. No hurry, maybe just copy your comment to an issue to keep your analysis available.

Reply:
All this is tracked in separate issues.